### PR TITLE
MLDB-1595 add count_distinct aggregator

### PR DESCRIPTION
**Pull request by steadtler**

### CODE REVIEW

File: sql/builtin_aggregators.cc [hunk 1]
```diff
@@ -720,6 +720,47 @@ struct CountAccum {
...
+    {
+       return ExpressionValue(knownValues.size(), ts);
+    }
+
+    void merge(DistinctAccum* src)
+    {
+        knownValues.insert(src->knownValues.begin(), src->knownValues.end());
+    }
+    
+    std::set<CellValue> knownValues;
```
Comment (mailletf) [inline sql/builtin_aggregators.cc]: unordered_set ?

File: container_files/public_html/doc/builtin/sql/ValueExpression.md [hunk 2]
```diff
@@ -486,6 +486,7 @@ The following standard SQL aggregation functions are supported. They may only be
 - `max` calculates the maximum of all values in the group.
 - `count` calculates the number of non-null values in the group.
     - `count(*)` is a special function which will count the number of rows in the group with non-null values in any column
+- 'count_distinct' calculates the number of unique, distinct non-null values in the group.
```
Comment (mailletf) [inline container_files/public_html/doc/builtin/sql/ValueExpression.md]: calculates -> counts ?

File: container_files/public_html/doc/builtin/sql/ValueExpression.md [hunk 3]
```diff
@@ -486,6 +486,7 @@ The following standard SQL aggregation functions are supported. They may only be
 - `max` calculates the maximum of all values in the group.
 - `count` calculates the number of non-null values in the group.
```
Comment (mailletf) [inline container_files/public_html/doc/builtin/sql/ValueExpression.md]: calculates -> counts?

File: testing/MLDB-1595-count-distinct.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        ds = mldb.create_dataset({ "id": "sample2", "type": "sparse.mutable" })
+        ds.record_row("a",[["x", 1, 0]])
+        ds.record_row("b",[["x", 2, 0]])
+        ds.record_row("c",[])
+        ds.record_row("d",[["x", 1, 0]])
+        ds.record_row("e",[["x", 3, 0]])
+        ds.commit()
+
+        res = mldb.query('''
+            select count_distinct({x as x, x % 2 as y}) as v from sample2
```
Comment (mailletf) [inline testing/MLDB-1595-count-distinct.py]: très cool comme feature!

**Comment by mailletf** [comment]
+1 with minor comments

File: container_files/public_html/doc/builtin/sql/ValueExpression.md (hunk 2 above)
Comment (steadtler) [inline container_files/public_html/doc/builtin/sql/ValueExpression.md]: All the other ones say calculates.

Comment (nicolaskruchten) [inline container_files/public_html/doc/builtin/sql/ValueExpression.md]: i'll do a pass on the documentation after :)

**Pull request merged**